vowpalwabbit/core/src/reductions/automl.cc

```diff
@@ -92,16 +92,65 @@ void finish_example(VW::workspace& all, automl<CMType>& data, VW::multi_ex& ec)
   VW::finish_example(all, ec);
 }
 
+template <typename CMType>
+void convert_to_single_model(VW::workspace& all, automl<CMType>& data)
+{
+  options_i& options = *all.options;
+  if (!options.was_supplied("predict_only_model")) { return; }
+  // Clear non-champ weights first
+
+  std::swap(*data.cm->_gd_normalized, data.cm->per_live_model_state_double[0]);
```

Comment:
The rest of this is generic and would work with other non-gd base reductions.  Would be good to fix gd and cb_adf so they save per model state for these things.  I realize that's big in scope so add an issue to the backlog?

Reply:
Two active PRs this one https://github.com/VowpalWabbit/vowpal_wabbit/pull/4023 blocked on search.cc bug https://github.com/VowpalWabbit/vowpal_wabbit/pull/4031